app/main.py

```diff
@@ -1,2 +1,39 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self._capacity = 8
+        self._storage = [None for _ in range(self._capacity)]
+        self._flow = 0
+
+    def __setitem__(self, key_, value_):
+        key_hash = hash(key_)
+        index = key_hash % self._capacity
+        while self._storage[index] is not None:
+            if self._storage[index][0] == key_hash:
+                self._flow -= 1
+                break
+            else:
+                index = (index + 1) % self._capacity
+        self._storage[index] = [key_hash, value_, key_]
+        self._flow += 1
+        if self._flow > self._capacity * 2 / 3:
+            self._refactor()
+
+    def __getitem__(self, key_):
+        key_hash = hash(key_)
+        index = key_hash % self._capacity
+        while self._storage[index] is not None:
+            if self._storage[index][0] == key_hash:
+                return self._storage[index][1]
+            index = (index + 1) % self._capacity
+
+    def __len__(self):
+        return self._flow
+
+    def _refactor(self):
+        self._capacity *= 2
+        self.old_storage = self._storage
+        self._storage = [None for _ in range(self._capacity)]
+        self._flow = 0
+        for i in self.old_storage:
```

Comment:
good solution, but I think, variable name "i" is not readable in this situation. Change name "i" on more informative, and then code stand more readable